pkgs/desktops/gnome-3/extensions/default.nix

```diff
@@ -0,0 +1,25 @@
+{pkgs, ...}: let
+  # Index of all scraped extensions (with supported versions)
+  extensionsIndex = builtins.fromJSON (builtins.readFile ./extensions.json);
+  # Filter the extensions index to the matching Gnome shell version.
+  filterShellVersion = shell-version: extensions-index:
+    (builtins.map (e: {
+      inherit (e) uuid name description link pname;
+      inherit (e.shell_version_map.${shell-version}) version sha256;
+    }) (builtins.filter
```

Comment:
You are nesting too many inline-defined operations here. I can read and understand it but we should optimize for readability.

Proposal:
Factor out the `(e: (builtins.hasAttr shell-version e."shell_version_map"))` part and put it in a let binding at the top of the function. Also give it a good name so we know what it does.